File: programming/c++/Loops.cpp
#include <iostream>
#include <vector>

int main()
{
    std::vector<std::string> vec
    {
        "Alfa",
        "Bravo",
        "Charlie"
    };

    //Loop through all item in a data structure (array can't)
    for (std::string name : vec)
    {
        std::cout << name << std::endl;
    }

    //Infinite Loop
    for (;;)
    {
        std::cout << "Never gonna let you down!" << std::endl;
    }
}

File: programming/rust/Loops.rs
fn main() {
	let arr: [&str; 3] = ["Alfa", "Bravo", "Charlie"];

	for name in &arr { // iterate through a container
		println!("{}", name);
	}

	let mut a = 0;
	while arr[a] != "Bravo" { // while loop
		println!("{}", arr[a]);
		a += 1;
	}

	loop { // infinite loop
		println!("Never gonna give you up");
	}
}